src/LegibleLambdas.jl

```diff
@@ -17,100 +17,20 @@ julia> @λ(x -> g(x)/3)
 """
 :(@λ)
 
-# NOTE: Base.@locals is in v1.1
-@static if VERSION < v"1.1.0"
-    macro λ(ex)
-        if ex.head == :(->)
-            ex_cut = ex |> (ex -> postwalk(cutlnn, ex)) |> (ex -> postwalk(cutblock, ex))
-            name = replace((repr(ex_cut)[2:end]), "->" => " -> ")
-            :(LegibleLambda($name, $(esc(ex))))
-        else
-            throw("Must be called on a Lambda expression")
-        end
-    end
-
-    struct LegibleLambda{F <: Function} <: Function
-        name::String
-        λ::F
-    end
-
-    (f::LegibleLambda)(args...) = (f.λ)(args...)
-
-    function Base.show(io::IO, f::LegibleLambda)
-        print(io, f.name)
-    end
-
-    function Base.show(io::IO, ::MIME"text/plain", f::LegibleLambda)
-        print(io, f.name)
-    end
-
-else
-
-    macro λ(ex)
-        if ex.head == :(->)
-            ex_cut = ex |> (ex -> postwalk(cutlnn, ex)) |> (ex -> postwalk(cutblock, ex))
-            name = (repr(ex_cut)[2:end])
-            return quote
-                LegibleLambda($(esc(ex)), $(QuoteNode(ex)), Base.@locals)
-            end
-        else
-            throw("Must be called on a Lambda expression")
-        end
-    end
-
-    struct LegibleLambda{F <: Function, T} <: Function
-        λ::F
-        ex::Expr
-        vars::Dict{Symbol, T}
-    end
-
-    (f::LegibleLambda)(args...; kwargs...) = (f.λ)(args...; kwargs...)
-
-    function to_string(f::LegibleLambda)
-        ex = f.ex
-        for (s, v) in f.vars
-            ex = postwalk(x->x === s ? v : x, ex)
-        end
-
-        ex_cut = ex |> (ex -> postwalk(cutlnn, ex)) |> (ex -> postwalk(cutblock, ex))
-        return replace((repr(ex_cut)[2:end]), "->" => " -> ")
-    end
-
-    function Base.show(io::IO, f::LegibleLambda)
-        print(io, to_string(f))
-    end
-
-    function Base.show(io::IO, ::MIME"text/plain", f::LegibleLambda)
-        print(io, to_string(f))
-    end
-
-end
-
 macro lambda(ex)
     return :(@λ($ex))
 end
```

Comment:
I'm not certain, but I think this should read
```julia
macro lambda(ex)
    return esc(:(@λ($ex)))
end
```
instead. This is subtle but it causes problems with things like 
```julia
julia> @lambda((x; y=1) -> x + y)
ERROR: syntax: "LegibleLambdas.x" is not a valid function argument name
```
whereas 
```julia
macro lambda2(ex)
        return esc(:(@λ($ex)))
end

julia> @lambda2((x; y=1) -> x + y)
(begin
          x
          y = 1
      end -> x + y)
```

Reply:
Ah, yeah, good gotcha